File: constellation.cpp
#include<bits/stdc++.h>
using namespace std;

map<queue<vector<char>>, char> vowels;

void initialize() 
{
    queue<vector<char>> v;
    // A
    v.push({'.', '*', '*'});
    v.push({'*', '*', '.'});
    v.push({'.', '*', '*'});
    vowels.insert({v, 'A'});
    while(!v.empty())
        v.pop();
    

    // E
    v.push({'*', '*', '*'});
    v.push({'*', '*', '*'});
    v.push({'*', '*', '*'});
    vowels.insert({v, 'E'});
    while(!v.empty())
        v.pop();

    // I
    v.push({'*', '.', '*'});
    v.push({'*', '*', '*'});
    v.push({'*', '.', '*'});
    vowels.insert({v, 'I'});
    while(!v.empty())
        v.pop();

    //O
    v.push({'*', '*', '*'});
    v.push({'*', '.', '*'});
    v.push({'*', '*', '*'});
    vowels.insert({v, 'O'});
    while(!v.empty())
        v.pop();

    //U
    v.push({'*', '*', '*'});
    v.push({'.', '.', '*'});
    v.push({'*', '*', '*'});
    vowels.insert({v, 'U'});
    while(!v.empty())
        v.pop();
}

int main()
{
    initialize();

    int n;
    cin >> n;

    vector<vector<char>> x(n, vector<char>(3));

   for(int i=0;i<3;i++)
  	{
    	for(int j=0;j<n;j++)	
  	    	{
                  cin>>x[j][i];    
                // cout << x[j][i] << " ";
            }
       // cout << "\n";
    }
    string constellation = "";
    queue<vector<char>> q;
    for ( int i=0; i<n; i++ )
    {
        if (q.size()==3)
            {
            if (vowels.find(q)!=vowels.end())
                {
                    constellation += vowels[q];
                    while(!q.empty())
                        q.pop();
                }
            else
            q.pop();   
            }
        if (x[i][0]=='#'&&x[i][1]=='#'&&x[i][2]=='#')
            {
                constellation += '#';
                while (!q.empty())
                    q.pop();
                continue;
            }
        q.push(x[i]);
    }
    
    if (q.size()==3&&vowels.find(q)!=vowels.end())
        constellation += vowels[q];              

    cout << constellation;

    return 0;
}

File: constellation.py
def initialize():
    q = []
    # A
    s = ""
    q.append(['.', '*', '*'])
    q.append(['*', '*', '.'])
    q.append(['.', '*', '*'])
    s = ''.join(map(str, q))
    vowels[s] = 'A'
    q.clear()

    #E
    q.append(['*', '*', '*'])
    q.append(['*', '*', '*'])
    q.append(['*', '*', '*'])
    s = ''.join(map(str, q))
    vowels[s] = 'E'
    q.clear()
    
    #I
    q.append(['*', '.', '*'])
    q.append(['*', '*', '*'])
    q.append(['*', '.', '*'])
    s = ''.join(map(str, q))
    vowels[s] = 'I'
    q.clear()

    #O
    q.append(['*', '*', '*'])
    q.append(['*', '.', '*'])
    q.append(['*', '*', '*'])
    s = ''.join(map(str, q))
    vowels[s] = 'O'
    q.clear()

    #U
    q.append(['*', '*', '*'])
    q.append(['.', '.', '*'])
    q.append(['*', '*', '*'])
    s = ''.join(map(str, q))
    vowels[s] = 'U'
    q.clear()

    return vowels

vowels = {}
vowels = initialize()
n = int(input())
x = []
for i in range(n):
    x.append(['.', '.', '.'])
for i in range(3):
    l = []
    l = list(input())
    for j in range(n):
        x[j][i] = l[j]
    

constellation = ""
star = []
for i in range(n):
    if len(star)==3:
        s = ''.join(map(str, star))
        if s in vowels:
            constellation += vowels[s]
            star.clear()
        else:
            star.pop(0)
    if x[i]==['#', '#', '#']: 
        star.clear()
        constellation += '#'
        continue
    star.append(x[i])
if len(star)==3:
    s = ''.join(map(str, star))
    if s in vowels:
        constellation += vowels[s]
        
print(constellation, end="")